sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -240,6 +240,54 @@ def create_node(self, node_config, tags, count):
 
     def terminate_node(self, node_id):
         logger.info(config.log_prefix + 'calling delete_namespaced_pod')
+        def find(l, predicate):
+            """
+            Utility function to find element in given list
+            """
+            results = [x for x in l if predicate(x)]
+            return results[0] if len(results) > 0 else None
+
+        sshjump_name = None
+        try:
+            sshjump_name = self.node_tags(node_id).get('skypilot-sshjump')
+        except kubernetes.api_exception() as e:
+            if e.status == 404:
+                logger.warning(config.log_prefix +
+                               f'Tried to get pod {node_id},'
+                               ' but the pod was not found (404).')
+            else:
+                raise
+        if sshjump_name:
+            try:
+                sshjump_pod = kubernetes.core_api().read_namespaced_pod(
+                    sshjump_name, self.namespace)
+                containersReadyCondition = find(
+                    sshjump_pod.status.conditions, lambda c: c.type == 'ContainersReady')
+                if containersReadyCondition and \
+                    containersReadyCondition.status == 'False':
+                    # The main container is not ready. To be on the safe-side
+                    # and prevent a dangling sshjump pod - lets remove it and
+                    # the service. Otherwise main container is ready and its lcm
+                    # takes care of the cleaning
+                    kubernetes.core_api().delete_namespaced_pod(
```

Comment:
Thanks @aviweit - a concern I have is this is splintering the ssh jump pod management code across modules. Jump pod creation is handled in `authentication.py::setup_kubernetes_authentication`, whereas this code is present in the node provider. Can we move the creation code also to node_provider?

Reply:
Update - I have done some refactoring in `k8s_cloud_beta1` with the above suggestion. Can you merge the latest changes into this branch?